include/fmt/std.h

```diff
@@ -277,10 +277,22 @@ FMT_END_NAMESPACE
 FMT_BEGIN_NAMESPACE
 
 FMT_EXPORT
+
+namespace detail {
+
+template <typename T, typename Char>
+struct is_expected_formattable : is_formattable<T, Char> {};
+
+template <typename Char>
+struct is_expected_formattable<void, Char> : std::true_type {};
+
+}  // namespace detail
+
 template <typename T, typename E, typename Char>
-struct formatter<std::expected<T, E>, Char,
-                 std::enable_if_t<is_formattable<T, Char>::value &&
-                                  is_formattable<E, Char>::value>> {
+struct formatter<
+    std::expected<T, E>, Char,
+    std::enable_if_t<detail::is_expected_formattable<T, Char>::value &&
```

Comment:
We can not use:
```c++
std::is_void<T>::value || is_formattable<T, Char>::value
```
since ``is_formattable<void, Char>::value`` give compile-time error.

@vitaut